app/src/main/java/com/example/firstlesson/ui/fragment/AllGoalsFragment.kt

```diff
@@ -0,0 +1,281 @@
+package com.example.firstlesson.ui.fragment
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.icu.text.SimpleDateFormat
+import android.location.Location
+import android.os.Build
+import android.os.Bundle
+import android.text.SpannableStringBuilder
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.firstlesson.R
+import com.example.firstlesson.adapter.AllGoalsAdapter
+import com.example.firstlesson.databinding.FragmentAllGoalsBinding
+import com.example.firstlesson.databinding.FragmentCreateGoalBinding
+import com.example.firstlesson.db.GoalsDatabase
+import com.example.firstlesson.entity.Goal
+import com.example.firstlesson.ui.MainActivity
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+import com.google.android.material.snackbar.Snackbar
+import com.google.android.material.timepicker.MaterialTimePicker
+import com.google.android.material.timepicker.TimeFormat
+import kotlinx.coroutines.*
+import java.util.*
+
+class AllGoalsFragment : Fragment() {
+
+    private lateinit var binding: FragmentAllGoalsBinding
```

Comment:
`Binding` classes should not have `lateinit` modifier, as it's leading to memory leaks, when `Fragment` outlives the `Binding` use scope.